### PR TITLE
Add panther_gazebo launch params docs

### PR DESCRIPTION
**Pull request by pkowalsk1**

### CODE REVIEW

File: panther_gazebo/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# panther_gazebo
+
+A package containing the launch files and dependencies needed to run the simulation in Ignition Fortress.
```
Comment (KmakD) [inline panther_gazebo/README.md]:
```suggestion
A package containing the launch files and dependencies needed to run the simulation in Gazebo Fortress.
```
I think this is the actual official name after whole renaming stuff.

File: panther_gazebo/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## Usage
+
+The recommended method for launching the simulation is by utilizing the [simulation.launch.py](./launch/simulation.launch.py) file. Below, you will find launch arguments that enable simulation configuration.
+
+### Launch Arguments
+
+- `battery_config_path` [*string*, default: **<panther_gazebo share directory>/config/battery_plugin_config.yaml**]: Path to the Ignition `LinearBatteryPlugin` configuration file. This configuration is intended for use in simulations only. For more information on how to configure this plugin, please refer to this [tab](#linear-battery-plugin).
```
Comment (KmakD) [inline panther_gazebo/README.md]:
```suggestion
- `battery_config_path` [*string*, default: **panther_gazebo/config/battery_plugin_config.yaml**]: Path to the Ignition `LinearBatteryPlugin` configuration file. This configuration is intended for use in simulations only. For more information on how to configure this plugin, please refer to the [Linear Battery Plugin](#linear-battery-plugin) section.
```
As for packages paths, this approach seems more clear. It should be obvious that it refers to package shared directory. Same applies for other params with package names in default value.

File: panther_gazebo/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+### Linear Battery Plugin
+
+It is possible to simulate the battery operation of the Panther robot. By default, this feature is disabled, but you can enable it by setting the `simulate_discharging` parameter to `true` in the `battery_plugin_config.yaml` file or in the file pointed to by the `battery_config_path` parameter. Below, you will find the plugin parameters that enable battery simulation.
+
+- `simulate_discharging` [*bool*, default: **false**]: Enables battery simulation. If set to `true`, the battery will discharge **at a constant rate** (regardless of joint torque), and if it depletes completely, the robot will stop moving. When set to `false`, the battery will not discharge, but the battery status information will still be published on the `/battery` topic.
+- `initial_charge_percentage` [*float*, default: **70.0**]: Sets the initial charge percentage of the battery.
```
Comment (KmakD) [inline panther_gazebo/README.md:32]: What will be the published value? Percentage should be in range 0 - 1. Not sure if this is wrong or there is some conversion on the way.

File: panther_gazebo/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+- `charging_time` [*float*, default: **6.0**]: Specifies the charging time for the battery in hours.
+- `power_load` [*float*, default: **120.0**]: Represents the average power load during normal operation **[W]** and is initially set to 120.0 W. With the default power load of 120.0 W, the robot can operate for up to 8 hours. When the `simulate_discharging` parameter is set to `false`, this value defaults to 0.0 W. Users are encouraged to customize this value to match their specific requirements. For more information on Panther power consumption, please refer to [Panther Battery & Charging Documentation](https://husarion.com/manuals/panther/#battery--charging).
+
+#### Charging Process
+
+Unfortunately, there is no straightforward way to exchange `LinearBatteryPlugin` services between ROS and Gazebo Transport, so you need to use the `ign` commands. As a result, the method of charging differs between the real and simulated robot.
```
Comment (KmakD) [inline panther_gazebo/README.md:42]: I'm missing info, how long does it take to charge the battery. Even if it's instant it's worth mentioning

File: panther_gazebo/README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+ign service --service /model/panther/battery/panther_battery/recharge/stop --reqtype ignition.msgs.Boolean --reptype ignition.msgs.Empty --req '' --timeout 0
+```
+
+### Use of GPS in Simulation
+
+The NavSat system is utilized to publish the Panther robot's position within the Gazebo world. It manages navigation satellite sensors, such as GPS, which report position and velocity in spherical coordinates (latitude/longitude) through Ignition Transport.
+
+The NavSat sensors depend on the spherical coordinates of the world origin being configured. This configuration can be accomplished, for instance, by employing the `<spherical_coordinates>` tag within the world's SDF or by utilizing the Ignition `/world/world_name/set_spherical_coordinates` service.
+
+To obtain GPS data in Ignition, all you need to do is include the [external_antenna](../panther_description/urdf/components/external_antenna.urdf.xacro) macro to your robot model and add the following tag to your world's SDF file and point this file with `world` param:
```
Comment (KmakD) [inline panther_gazebo/README.md]: Missing info that it is also required to modify gz_bridge to sent NavSat topic. Also, I would add info that default world is already configured to support GPS data, If user want's to use custom world then he needs to add spherical_coordinates.

File: panther_gazebo/README.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+```bash
+ign service --service /model/panther/battery/panther_battery/recharge/stop --reqtype ignition.msgs.Boolean --reptype ignition.msgs.Empty --req '' --timeout 0
+```
+
+### Use of GPS in Simulation
+
+The NavSat system is utilized to publish the Panther robot's position within the Gazebo world. It manages navigation satellite sensors, such as GPS, which report position and velocity in spherical coordinates (latitude/longitude) through Ignition Transport.
+
+The NavSat sensors depend on the spherical coordinates of the world origin being configured. This configuration can be accomplished, for instance, by employing the `<spherical_coordinates>` tag within the world's SDF or by utilizing the Ignition `/world/world_name/set_spherical_coordinates` service.
```
Comment (KmakD) [inline panther_gazebo/README.md]:
```suggestion
The NavSat sensors requires the spherical coordinates of the world origin to be configured. This configuration can be accomplished, for instance, by employing the `<spherical_coordinates>` tag within the world's SDF or by utilizing the Ignition `/world/world_name/set_spherical_coordinates` service.
```

File: panther_gazebo/README.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+A package containing the launch files and dependencies needed to run the simulation in Ignition Fortress.
+
+## Usage
+
+The recommended method for launching the simulation is by utilizing the [simulation.launch.py](./launch/simulation.launch.py) file. Below, you will find launch arguments that enable simulation configuration.
+
+### Launch Arguments
+
+- `battery_config_path` [*string*, default: **<panther_gazebo share directory>/config/battery_plugin_config.yaml**]: Path to the Ignition `LinearBatteryPlugin` configuration file. This configuration is intended for use in simulations only. For more information on how to configure this plugin, please refer to this [tab](#linear-battery-plugin).
+- `controller_config_path` [*string*, default: **<panther_controller share directory>/config/<wheel_type arg>_controller.yaml**]: Path to the controller configuration file. If you want to use a custom configuration, you can specify the path to your custom controller configuration file here.
```
Comment (KmakD) [inline panther_gazebo/README.md]:
```suggestion
- `controller_config_path` [*string*, default: **panther_controller/config/<wheel_type arg>_controller.yaml**]: Path to the controller configuration file. If you want to use a custom configuration, you can specify the path to your custom controller configuration file here.
```

File: panther_gazebo/README.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+- `gz_bridge_config_path` [*string*, default: **<panther_gazebo share directory>/config/gz_bridge.yaml**]: Path to the `parameter_bridge` configuration file. For detailed information on configuring the `parameter_bridge`, please refer to this [example](https://github.com/gazebosim/ros_gz/tree/ros2/ros_gz_bridge#example-5-configuring-the-bridge-via-yaml).
```
Comment (KmakD) [inline panther_gazebo/README.md]:
```suggestion
- `gz_bridge_config_path` [*string*, default: **panther_gazebo/config/gz_bridge.yaml**]: Path to the `parameter_bridge` configuration file. For detailed information on configuring the `parameter_bridge`, please refer to this [example](https://github.com/gazebosim/ros_gz/tree/ros2/ros_gz_bridge#example-5-configuring-the-bridge-via-yaml).
```

File: panther_gazebo/README.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+- `pos_x` [*float*, default: **5.0**]: spawn position **[m]** of the robot in the world in **X** direction.
+- `pos_y` [*float*, default: **-5.0**]: spawn position **[m]** of the robot in the world in **Y** direction.
+- `pos_z` [*float*, default: **0.2**]: spawn position **[m]** of the robot in the world in **Z** direction.
+- `rot_yaw` [*float*, default: **0.0**]: spawn yaw angle **[rad]** of the robot in the world.
+- `publish_robot_state` [*bool*, default: **true**]: Whether to launch the robot_state_publisher node. When set to `false`, users should publish their own robot description.
+- `wheel_config_path` [*string*, default: **<panther_description share directory>/config/<wheel_type arg>.yaml**]: Path to the wheel configuration file. If you want to use a custom configuration, you can specify the path to your custom wheel configuration file here. Please refer to the `wheel_type` parameter description for more information.
```
Comment (KmakD) [inline panther_gazebo/README.md]:
```suggestion
- `wheel_config_path` [*string*, default: **panther_description/config/<wheel_type arg>.yaml**]: Path to the wheel configuration file. If you want to use a custom configuration, you can specify the path to your custom wheel configuration file here. Please refer to the `wheel_type` parameter description for more information.
```

**Review by KmakD — CHANGES_REQUESTED** [review]

File: panther_gazebo/README.md (hunk 2 above)
Comment (pkowalsk1) [reply panther_gazebo/README.md]: As a result of our team's discussion, I reluctantly agree, but it seems to me that I will one day say, "Didn't I tell you?" :p

File: panther_gazebo/README.md (hunk 3 above)
Comment (pkowalsk1) [reply panther_gazebo/README.md:32]: Published value is:

```bash
percentage: 70.0
```

This value corresponds to the initial_charge_percentage. I suggest adding a brief explanation regarding this difference.

File: panther_gazebo/README.md (hunk 4 above)
Comment (pkowalsk1) [reply panther_gazebo/README.md:42]: There is a parameter named `charging_time`. The duration is determined by the user.

File: panther_gazebo/README.md [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+- `initial_charge_percentage` [*float*, default: **70.0**]: Sets the initial charge percentage of the battery.
+- `charging_time` [*float*, default: **6.0**]: Specifies the charging time for the battery in hours.
+- `power_load` [*float*, default: **120.0**]: Represents the average power load during normal operation **[W]** and is initially set to 120.0 W. With the default power load of 120.0 W, the robot can operate for up to 8 hours. When the `simulate_discharging` parameter is set to `false`, this value defaults to 0.0 W. Users are encouraged to customize this value to match their specific requirements. For more information on Panther power consumption, please refer to [Panther Battery & Charging Documentation](https://husarion.com/manuals/panther/#battery--charging).
+
+> **Note**
+>
+> The `percentage` field has a range of 0.0-100.0. This differs from the real functioning of Panthera, where, in accordance with the BatteryState message definition, the value is within the range of 0.0-1.0.
```
Comment (KmakD) [inline panther_gazebo/README.md]:
```suggestion
> The `percentage` field has a range of 0.0-100.0. This differs from the real functioning of Panther, where, in accordance with the BatteryState message definition, the value is within the range of 0.0-1.0.
```

**Review by KmakD — CHANGES_REQUESTED** [review]

**Review by KmakD — APPROVED** [review]

**Pull request merged**